packages/components/src/shared/src/useDisableInteraction.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * useDisableInteraction provides the original event handler but disables interaction
+ * if the boolean passed is true.
+ */
+import { useCallback, MouseEventHandler } from "react";
+
+export function useDisableClick<T = Element>(disabled?: boolean, handleClick?: MouseEventHandler<T>) {
```

Comment:
Are those parameters really optional? I would expect the caller to always provide those